src/routers/user.ts

```diff
@@ -5,47 +5,63 @@ import {
   userBodyValidatorOnUpdate,
   paramsIdValidator,
   userQuerySubstringLimitValidator,
+  userBodyCredentialsValidator,
 } from '../validation/validators';
 import { User } from '../models/typeORMModels';
 import { ValidatedRequest } from 'express-joi-validation';
 import {
   CreateUserBodySchema,
+  GetUserByCredentialsBodySchema,
   ParamsIDSchema,
   QuerySubstringLimitSchema,
   UpdateUserBodySchema,
 } from '../validation/types';
 import { methodLoggerMiddleware } from '../middlewares/methodLoggerMiddleware';
+import * as jwt from 'jsonwebtoken';
+import { checkTokenMiddleware } from '../middlewares/checkTokenMiddleware';
 
 const user = express.Router();
 const userService = new UserService();
 
 user.post(
   '/user',
-  userBodyValidatorOnCreate, methodLoggerMiddleware,
-  async (req: ValidatedRequest<CreateUserBodySchema>, res: Response, next: NextFunction) => {
+  userBodyValidatorOnCreate,
+  methodLoggerMiddleware,
+  checkTokenMiddleware,
```

Comment:
Same here re: adding middleware to app.use(), see groups router.

Reply:
fixed